### PR TITLE
Adding sample code for exporting specific datasets

### PR DESCRIPTION
**Pull request by kwwaikar**
*Issue #, if available:*
Adding sample code for exporting specific datasets
*Description of changes:*
Adding sample code for exporting specific datasets

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: subscribers/python/download-datasets/export-dataset.py [hunk 1]
```diff
@@ -0,0 +1,101 @@
...
+import re 
+import time
+import click
+import uuid
+import json
+
+dx = boto3.client('dataexchange')
+s3 = boto3.client('s3')
+
+#This function returns revision_ids corresponding to the dataset-id specified.
```
Comment (melchii) [inline subscribers/python/download-datasets/export-dataset.py:14]: nit: Data Set is two words (applies everywhere)

File: subscribers/python/download-datasets/export-dataset.py [hunk 2]
```diff
@@ -0,0 +1,101 @@
...
+    #Paginate and extract all revisions corresponding to the dataset specified.
+    revisions = []
+    res = dx.list_data_set_revisions(DataSetId=data_set_id)
+    next_token = res.get('NextToken')
+    revisions += res.get('Revisions')
+    while next_token:
+        res = dx.list_data_set_revisions(DataSetId=data_set_id,
+                                         NextToken=next_token)
+        revisions += res.get('Revisions')
+        next_token = res.get('NextToken')
```
Comment (melchii) [inline subscribers/python/download-datasets/export-dataset.py:26]: Simplify with do while statement?

File: subscribers/python/download-datasets/export-dataset.py [hunk 3]
```diff
@@ -0,0 +1,101 @@
...
+
+    #Extract IDs corresponding to the revision
+    revision_ids = []
+    for rev in revisions:
+        revision_ids.append(rev['Id'])
+    #print(revision_ids)
```
Comment (melchii) [inline subscribers/python/download-datasets/export-dataset.py:32]: Remove?

File: subscribers/python/download-datasets/export-dataset.py [hunk 4]
```diff
@@ -0,0 +1,101 @@
...
+    revisions += res.get('Revisions')
+    while next_token:
+        res = dx.list_data_set_revisions(DataSetId=data_set_id,
+                                         NextToken=next_token)
+        revisions += res.get('Revisions')
+        next_token = res.get('NextToken')
+
+    #Extract IDs corresponding to the revision
+    revision_ids = []
+    for rev in revisions:
```
Comment (melchii) [inline subscribers/python/download-datasets/export-dataset.py:30]: rev -> revision

**Review by melchii — CHANGES_REQUESTED** [review]

File: subscribers/python/download-datasets/README.md [hunk 5]
```diff
@@ -0,0 +1,30 @@
+# Download all Entitled Assets (Python)
```
[the remark this answers is not in the document]
Comment (kwwaikar) [reply subscribers/python/download-datasets/README.md]: Renamed to Export entitled data set assets

File: subscribers/python/download-datasets/README.md [hunk 6]
```diff
@@ -0,0 +1,30 @@
...
+
+This example accepts dataset_id(s),bucket,region, and exports all revisions corresponding to dataset-ids specified into an S3 bucket.
```
[the remark this answers is not in the document]
Comment (kwwaikar) [reply subscribers/python/download-datasets/README.md]: fixed

File: subscribers/python/download-datasets/README.md [hunk 7]
```diff
@@ -0,0 +1,30 @@
...
+Set AWS access key and secret.
+
+```
+$ export AWS_ACCESS_KEY_ID=<your-access-key-id>
+$ export AWS_SECRET_ACCESS_KEY=<your-secret-access-key>
+```
+
+The following policies are required for this AWS user.
+
+* AmazonS3FullAccess
```
[the remark this answers is not in the document]
Comment (kwwaikar) [reply subscribers/python/download-datasets/README.md]: Changed DataExchangeFullAccess to DataExchangeSubscriberFullAccess. Yes, subscriber full access policy does not include S3 permissions

File: subscribers/python/download-datasets/README.md [hunk 8]
```diff
@@ -0,0 +1,30 @@
...
+* AWSDataExchangeFullAccess
```
[the remark this answers is not in the document]
Comment (kwwaikar) [reply subscribers/python/download-datasets/README.md]: Fixed.

File: subscribers/python/download-datasets/export-dataset.py (hunk 1 above)
Comment (kwwaikar) [reply subscribers/python/download-datasets/export-dataset.py]: Fixed.

File: subscribers/python/download-datasets/export-dataset.py (hunk 2 above)
Comment (kwwaikar) [reply subscribers/python/download-datasets/export-dataset.py]: Skipped due to lack of native support for do-while in python.

File: subscribers/python/download-datasets/export-dataset.py (hunk 3 above)
Comment (kwwaikar) [reply subscribers/python/download-datasets/export-dataset.py]: fixed.

File: subscribers/python/download-datasets/export-dataset.py (hunk 4 above)
Comment (kwwaikar) [reply subscribers/python/download-datasets/export-dataset.py]: Fixed.

File: subscribers/python/export-data-sets/export-data-sets.py [hunk 9]
```diff
@@ -0,0 +1,101 @@
...
+        #Wait for all import jobs to finish
+        for job in job_ids:
+            max_time = time.time() + 60*60 # 1 hour
+            #print(job)
+            while time.time() < max_time :
+                response = dx.get_job(JobId=job_id);
+                status = response['State']
+                print('STATUS: ',job,'get_job_status'+": {}".format(status))
+                if status == "COMPLETED" or status == "ERROR":
+                    break
```
Comment (melchii) [inline subscribers/python/export-data-sets/export-data-sets.py:69]: This will complete successfully if all jobs fail. Is that desired?

**Pull request merged**